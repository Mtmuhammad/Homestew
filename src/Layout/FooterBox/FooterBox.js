import React from "react";
import "./FooterBox.scss"

const FooterBox = ({ img }) => {
  return (
    <div className="col-6 col-sm-4 col-md-2">
      <img
        className="img-fluid instagram-post"
        src={img}
        alt="Instagram post"
      />
    </div>
  );
};

export default FooterBox;
