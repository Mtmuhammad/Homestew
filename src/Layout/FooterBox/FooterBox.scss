@use "../../Styles/abstracts/mixins" as *;
@use "../../Styles/abstracts/variables" as *;

.instagram-post {
  height: 300px;
  margin-bottom: 10px;
  border-radius: 5px;

  @include respond(land) {
    height: 280px;
  }
  @include respond(port) {
    height: 240px;
  }
  @include respond(sport) {
    height: 300px;
  }
  @include respond(sphone) {
    height: 255px;
  }
  @include respond(xsphone) {
    height: 200px;
  }
  @include respond(tiny) {
    height: 170px;
  }
}
