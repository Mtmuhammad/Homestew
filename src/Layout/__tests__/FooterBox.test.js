import React from "react";
import { render, screen } from "@testing-library/react";
import FooterBox from "../FooterBox/FooterBox";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<FooterBox img={"Instagram.1jpg"}/>);
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<FooterBox img={"Instagram.1jpg"}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});

test("should render img", () => {
   render(<FooterBox img={"Instagram.1jpg"}/>)
   const img = screen.getByAltText(/instagram post/i)
   expect(img).toBeInTheDocument()
   expect(img).toHaveClass("img-fluid")
})