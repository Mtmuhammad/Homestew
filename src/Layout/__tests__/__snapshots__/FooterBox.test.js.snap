// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-6 col-sm-4 col-md-2"
  >
    <img
      alt="Instagram post"
      class="img-fluid instagram-post"
      src="Instagram.1jpg"
    />
  </div>
</DocumentFragment>
`;
